sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs

```diff
@@ -874,6 +935,228 @@ public async Task GetBatchSettingWithAllFields()
+            }
+        }
+
+        [Test]
+        public async Task GetBatchSettingContains()
+        {
+            ConfigurationClient service = GetClient();
+            ConfigurationSetting testSetting = CreateSetting("yzabcde", "Contains abc", "yzabcde");
+
+            try
+            {
+                await service.SetAsync(testSetting);
+
+                var selector = new SettingSelector("*abc*");
+
+                ConfigurationSetting[] settings = (await service.GetSettingsAsync(selector, CancellationToken.None).ToEnumerableAsync()).ToArray();
+
+                // There should be at least one key available.
+                Assert.GreaterOrEqual(settings.Length, 1);
+
+                foreach (ConfigurationSetting setting in settings)
+                {
+                    StringAssert.Contains("abc", setting.Key);
+                }
+            }
+            finally
+            {
+                await service.DeleteAsync(testSetting.Key);
+            }
+        }
+
+        [Test]
+        public async Task GetBatchSettingsWithCommaInSelectorKey()
+        {
+            ConfigurationClient service = GetClient();
+
+            ConfigurationSetting abcSetting = new ConfigurationSetting("ab,cd", "comma in key");
+            ConfigurationSetting xyzSetting = new ConfigurationSetting("wx,yz", "comma in key");
+
+            try
+            {
+                await service.SetAsync(abcSetting);
+                await service.SetAsync(xyzSetting);
+
+                var selector = new SettingSelector("ab,cd");
+                selector.Keys.Add("wx,yz");
+
+                ConfigurationSetting[] settings = (await service.GetSettingsAsync(selector, CancellationToken.None).ToEnumerableAsync()).ToArray();
+
+                Assert.GreaterOrEqual(settings.Length, 2);
+                Assert.IsTrue(settings.Any(s => s.Key == "ab,cd"));
+                Assert.IsTrue(settings.Any(s => s.Key == "wx,yz"));
+            }
+            finally
+            {
+                await service.DeleteAsync(abcSetting.Key);
+                await service.DeleteAsync(xyzSetting.Key);
+            }
+        }
+
+        [Test]
+        public async Task GetBatchSettingsWithCommaInSelectorKeyDoesNotOr()
+        {
+            ConfigurationClient service = GetClient();
+
+            ConfigurationSetting abcSetting = new ConfigurationSetting("abc", "abc setting");
+            ConfigurationSetting xyzSetting = new ConfigurationSetting("xyz", "xyz setting");
+
+            try
+            {
+                await service.SetAsync(abcSetting);
+                await service.SetAsync(xyzSetting);
+
+                var selector = new SettingSelector($"{abcSetting.Key},{xyzSetting.Key}");
+
+                ConfigurationSetting[] settings = (await service.GetSettingsAsync(selector, CancellationToken.None).ToEnumerableAsync()).ToArray();
+
+                Assert.GreaterOrEqual(settings.Length, 0);
```

Comment:
This assert will always succeed. Did you mean `CollectionAssert.IsEmpty(settings);`?